wear/src/main/java/io/homeassistant/companion/android/phone/PhoneSettingsListener.kt

```diff
@@ -135,6 +149,28 @@ class PhoneSettingsListener : WearableListenerService(), DataClient.OnDataChange
             val deviceName = dataMap.getString("DeviceName")
             val deviceTrackingEnabled = dataMap.getBoolean("LocationTracking")
             val notificationsEnabled = dataMap.getBoolean("Notifications")
+            val tlsClientCertificateData = dataMap.getByteArray("TLSClientCertificateData")
+            val tlsClientCertificatePassword = dataMap.getString("TLSClientCertificatePassword").orEmpty().toCharArray()
+
+            // load TLS key
+            if (tlsClientCertificateData != null && tlsClientCertificateData.isNotEmpty()) {
+                KeyStore.getInstance("PKCS12").apply {
+                    try {
+                        load(tlsClientCertificateData.inputStream(), tlsClientCertificatePassword)
+
+                        val alias = aliases().nextElement()
+                        val certificateChain = getCertificateChain(alias).filterIsInstance<X509Certificate>().toTypedArray()
+                        val privateKey = getKey(alias, tlsClientCertificatePassword) as PrivateKey
+
+                        // we store the TLS Client key under a static alias because there is currently
+                        // no way to ask the user for the correct alias
+                        keyStore.setData("TLSClientCertificate", privateKey, certificateChain)

```

Comment:
`TLSClientCertificate` should be a constant from the KeyStore, maybe even internal to that class.

Reply:
I added a public constant `KeyStoreImpl.ALIAS`. An internal constant does not work with the current KeyChain interface.